tests/test_meta.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import unicode_literals, absolute_import, division
+
+import re
+
+import pytest
+
+from pep517 import meta
+
+
+pep517_needs_python_3 = pytest.mark.xfail(
```

Comment:
what 🤔 why does this requires python 3? 🤔

Reply:
`pep517` is built with [flit](https://pypi.org/project/flit), which is a Python 3 only package. Because these tests are relying on this project's own repo/metadata to run the tests, those tests therefore have the same constraint. It's the same reason you can't install pep517 from source on Python 2.7:

```
$ python2.7 -m pip-run --no-binary=:all: --no-deps pep517                                                                                                                    
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting pep517
  Downloading https://files.pythonhosted.org/packages/79/27/1e6a8ba83dab2b243e1020e340552c129b0076fbb5518e0c2a9c44c7d056/pep517-0.5.0.tar.gz
  Installing build dependencies ... error
  Complete output from command /usr/local/opt/python@2/bin/python2.7 /usr/local/lib/python2.7/site-packages/pip install --ignore-installed --no-user --prefix /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-build-env-jCiSaR/overlay --no-warn-script-location --no-binary :all: --only-binary :none: -i https://pypi.org/simple -- flit:
  DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
  Collecting flit
    Could not find a version that satisfies the requirement flit (from versions: )
  No matching distribution found for flit
  
  ----------------------------------------
Command "/usr/local/opt/python@2/bin/python2.7 /usr/local/lib/python2.7/site-packages/pip install --ignore-installed --no-user --prefix /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-build-env-jCiSaR/overlay --no-warn-script-location --no-binary :all: --only-binary :none: -i https://pypi.org/simple -- flit" failed with error code 1 in None
```

These tests could be re-written to use mocks instead of the local pep517 checkout, but I'm satisfied with this approach.

Reply:
Should not be that hard to run the tests on a setuptools simple project, not 🤔 flit does generate wheel/sdist that can be installed on Python 2 🤔

Reply:
I do think it would be worthwhile for coverage reasons to retrieve metadata from a non PEP 517 source package.

> flit does generate sdist that can be installed on Python 2

Can you demonstrate this behavior? Because the pep517 sdist seems not to have that feature.

Reply:
Eh, looking through the CI and tests there's no such tests or coverage report yet. As such for now probably not worth it. Let's release and a subsequent PR can improve the CI to generate coverage, and then we can worry about this.